File: BOJ(Java 공부용)/10872.java
import java.io.*;
public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        String arr[] = br.readLine().split(" ");
        int n= Integer.parseInt(arr[0]);
        bw.write(String.valueOf(go(n)));
        bw.flush();
    }

    private static int go(int n) {
        if(n==0) return 1;
        return n*go(n-1);
    }
}

File: BOJ(Kotlin)/10872.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*
import kotlin.math.max
import kotlin.math.min

fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val st = StringTokenizer(br.readLine())
    val n = st.nextToken().toInt()
    val res = factorial(n)
    bw.write(res.toString())
    bw.flush()
    bw.close()
}
fun factorial(fact : Int) : Int {
    if (fact == 0) return 1
    else return fact * factorial(fact-1)
}